src/flag_gems/ops/repeat_interleave.py

```diff
@@ -53,3 +57,41 @@ def repeat_interleave_self_int(inp, repeats, dim=None, *, output_size=None):
     ndim = len(out_view_shape)
     copy_func.instantiate(ndim)(in_view, out0=out_view)
     return output
+
+
+@triton.jit
+def repeat_interleave_tensor_kernel(inputs_ptr, repeats_ptr, cumsum_ptr, out_ptr, size):
+    pid = tl.program_id(0)
+    mask = pid < size
+    cumsum = tl.load(cumsum_ptr + pid, mask, other=0)
+    inputs = tl.load(inputs_ptr + pid + 1, mask)
+    repeats = tl.load(repeats_ptr + pid + 1, mask)
+
+    tl.device_assert(repeats >= 0, "repeats can not be negative")
+
+    out_offsets = cumsum
+    for _ in range(repeats):
+        tl.store(out_ptr + out_offsets, inputs)
+        out_offsets += 1
+
```

Comment:
Maybe write out by tile instead of by element.

something like

```python
    out_ptr += out_offset
    for start_k in range(0, repeats, 32):
        offsets_k = start_k + tl.arange(0, 32)
        mask_k = offsets_k < repeats
        tl.store(out_ptr + offsets_k, pid, mask=mask_k)
```

Reply:
Done. Retesting the performance on NV-A100, compared to the previous implementation, we've made some improvements.

```
benchmark/test_pointwise_perf.py::test_perf_repeat_interleave_tensor 
Operator repeat_interleave_tensor Performance Test (dtype=torch.int32, mode=cuda)
Size    Torch Latency (ms)    Gems Latency (ms)    Gems Speedup
---------------------------------------------------------------
1024              0.095232             0.139264           0.684
6144              0.093184             0.157696           0.591
11264             0.093184              0.15872           0.587
16384             0.095232              0.15872             0.6
21504             0.100352             0.159744           0.628
26624               0.1024             0.159744           0.641
31744             0.104448             0.161792           0.646
36864             0.104448             0.164864           0.634
41984             0.104448              0.16384           0.638
47104             0.104448             0.164864           0.634
52224             0.104448             0.165888            0.63
57344             0.104448             0.165888            0.63
62464             0.104448              0.17408             0.6
67584             0.105472             0.176128           0.599
72704             0.105472             0.181248           0.582
77824              0.10752             0.182272            0.59
PASSED
```